core/src/test/kotlin/org/jetbrains/kotlinx/dataframe/api/simplify.kt

```diff
@@ -0,0 +1,37 @@
+package org.jetbrains.kotlinx.dataframe.api
+
+import io.kotest.matchers.shouldBe
+import org.jetbrains.kotlinx.dataframe.DataRow
+import org.jetbrains.kotlinx.dataframe.alsoDebug
+import org.jetbrains.kotlinx.dataframe.columns.ColumnSet
+import org.jetbrains.kotlinx.dataframe.columns.ColumnWithPath
+import org.jetbrains.kotlinx.dataframe.columns.SingleColumn
+import org.jetbrains.kotlinx.dataframe.columns.asColumnSet
+import org.jetbrains.kotlinx.dataframe.impl.columns.allColumnsExceptKeepingStructure
+import org.jetbrains.kotlinx.dataframe.impl.columns.singleOrNullImpl
+import org.jetbrains.kotlinx.dataframe.impl.columns.transformSingleWithContext
+import org.jetbrains.kotlinx.dataframe.impl.columns.transformWithContext
+import org.jetbrains.kotlinx.dataframe.samples.api.age
+import org.jetbrains.kotlinx.dataframe.samples.api.firstName
+import org.jetbrains.kotlinx.dataframe.samples.api.secondName
+import org.jetbrains.kotlinx.dataframe.samples.api.lastName
+import org.jetbrains.kotlinx.dataframe.samples.api.name
+import org.junit.Test
+
+class SimplifyTests : ColumnsSelectionDslTests() {
+
+    @Test
+    fun simplify() {
+        df.select { cols(name.firstName, name.lastName, age).simplify() } shouldBe
```

Comment:
Honestly, only learning these tests I started to understand what this function does. Could we add more complex cases here? Especially with repeated subcolumns, for example?